### PR TITLE
fix(updater): change off the install cwd before the Windows restart-to-update swap

### PR DESCRIPTION
**Pull request by Jsakkos**
## Summary

Restart-to-update has failed on Windows across many attempts. Systematic debugging (root cause confirmed empirically, not guessed) found that **download/unpack was never the problem** — the failure was entirely in the self-overwrite swap, and the crash-safe robocopy swap added in #322 **never actually completed once**.

**Root cause:** `_spawn_detached_helper` launched the detached `cmd /c engram_update.bat` with **no `cwd=`**, so it inherited engram's working directory — the install dir for a double-clicked onedir exe. On Windows a directory that is any process's current directory cannot be renamed, so the helper's own first swap step `move "%INSTALL%" "%OLDDIR%"` failed with *"being used by another process"* → `:fail_no_swap` → rollback → **update silently never applied, no UI error, on every attempt.** (The absence of `~/.engram/update_helper.log` despite 6 logged helper launches was because those ran the even-older xcopy-in-place bat, which wrote no log.)

## Evidence

Reproduced and then verified in an isolated swap harness that runs the **real** `_render_update_bat` against throwaway copies, A/B-testing the helper's cwd (never touches the real install):

| Run | helper cwd | `move install -> .old` | Before fix | After fix |
|-----|-----------|------------------------|-----------|-----------|
| A | == install dir (production) | `0 dir(s) moved` / *used by another process* | rolled back | **success** |
| B | neutral dir | `1 dir(s) moved` | success | success |

robocopy succeeds in both; the only variable is cwd.

## Fix

`backend/app/core/updater.py`:
- `_spawn_detached_helper` passes `cwd=tempfile.gettempdir()` to both Popen calls.
- `_render_update_bat` does `cd /d "%TEMP%"` before the swap (defense in depth).
- Relaunch via `start "" /D "%INSTALL%"` so the new app keeps the install dir as cwd (matches a normal double-click; frozen DB is absolute `~/.engram/engram.db` regardless).
- Captures robocopy's exit into `%RC%` **before** logging it (a stray `echo` resets `%ERRORLEVEL%`, which would have made every copy look successful).
- Logs the inherited `%CD%`, all resolved paths, the robocopy exit, and a `done` marker on every terminal branch — so a single `update_helper.log` self-diagnoses any future edge case.

## Tests

- `tests/unit/test_updater.py`: added failing-first tests for the neutral spawn cwd (both spawn paths), the bat `cd` off the install dir before the swap, the diagnostic context, and the `done` markers. 40 unit + 6 integration update tests pass; ruff clean.

## ⚠️ Chicken-and-egg (important)

The swap is performed by the **currently-running** binary, so this fix only takes effect once a user is running a build that *contains* it. A user on 0.15.4 updating to the next release still runs 0.15.4's buggy helper, so that jump still rolls back — existing installs need **one more manual download** to a fixed build, after which restart-to-update works for good. This is the standing argument for the deferred side-by-side-launcher re-architecture.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/app/core/updater.py [hunk 1]
```diff
@@ -668,6 +673,11 @@
             f'set "LOG={log_path}"',
             f'set "EXE=%INSTALL%\\{exe}"',
             f'echo [engram-update] start (pid {pid}) > "%LOG%"',
+            # Diagnostic context: %CD% here is the cwd we INHERITED from engram. If it
+            # equals %INSTALL%, the `cd /d` below is what saves the swap (see there).
+            'echo [engram-update] cwd=%CD% >> "%LOG%"',
+            "echo [engram-update] INSTALL=%INSTALL% NEWDIR=%NEWDIR% OLDDIR=%OLDDIR%"
+            ' SRC=%SRC% EXE=%EXE% >> "%LOG%"',
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/app/core/updater.py]: Fixed in 4dcf7fa — split the two adjacent string literals into two separate `echo` statements (each a complete bat line), so there's no implicit concatenation in the list.
